base/applications/network/ipconfig/ipconfig.c

```diff
@@ -238,6 +239,75 @@ PTCHAR PrintMacAddr(PBYTE Mac)
 }
 
 
+/* convert time_t to localized string */
+_Ret_opt_z_ PTSTR AllocateLocalizedDateTimeFromTimet(_In_ time_t TimeStamp)
+{
+    struct tm* ptm;
+    SYSTEMTIME SystemTime;
+    INT DateCchSize, TimeCchSize, TotalCchSize, i;
+    PTSTR DateTimeString, psz;
+
+    /* Convert Unix time to SYSTEMTIME */
+    /* FIXME: localtime_s may preferred but the declaration not found here */
+    ptm = localtime(&TimeStamp);
+    if (!ptm)
+    {
+        return NULL;
+    }
+    SystemTime.wYear = ptm->tm_year + 1900;
+    SystemTime.wMonth = ptm->tm_mon + 1;
+    SystemTime.wDay = ptm->tm_mday;
+    SystemTime.wHour = ptm->tm_hour;
+    SystemTime.wMinute = ptm->tm_min;
+    SystemTime.wSecond = ptm->tm_sec;
+
+    /* Get total size in characters required of buffer */
+    DateCchSize = GetDateFormat(LOCALE_USER_DEFAULT, 0, &SystemTime, NULL, NULL, 0);
+    if (!DateCchSize)
+    {
+        return NULL;
+    }
+    TimeCchSize = GetTimeFormat(LOCALE_USER_DEFAULT, 0, &SystemTime, NULL, NULL, 0);
+    if (!TimeCchSize)
+    {
+        return NULL;
+    }
+    /* Two terminating null are included, the first one will be replaced by space */
+    TotalCchSize = DateCchSize + TimeCchSize;
+
+    /* Allocate buffer and format time string */
+    DateTimeString = (PTSTR)HeapAlloc(ProcessHeap, 0, TotalCchSize * sizeof(TCHAR));
+    if (!DateTimeString)
+    {
+        return NULL;
+    }
+
+    /* Get date string */
+    i = GetDateFormat(LOCALE_USER_DEFAULT, 0, &SystemTime, NULL, DateTimeString, TotalCchSize);
+    if (!i)
```

Comment:
I would much prefer you swapping this condition and next 2 levels (here and below), instead of using goto.

Reply:
commited